src/state/slices/tradeQuoteSlice/selectors.ts

```diff
@@ -505,57 +506,35 @@ export const selectQuoteSellAmountUserCurrency = createSelector(
   },
 )
 
-export const selectActiveQuoteAffiliateBps: Selector<ReduxState, string | undefined> =
-  createSelector(selectActiveQuote, activeQuote => {
-    if (!activeQuote) return
+export const selectActiveQuoteAffiliateBps: Selector<ReduxState, string> = createSelector(
+  selectActiveQuote,
+  activeQuote => {
+    if (!activeQuote) return '0'
     return activeQuote.affiliateBps
-  })
-
-export const selectTradeQuoteAffiliateFeeAfterDiscountUsd = createSelector(
-  (state: ReduxState) =>
-    selectCalculatedFees(state, {
-      feeModel: 'SWAPPER',
-      inputAmountUsd: selectQuoteSellAmountUsd(state),
-    }),
-  selectActiveQuoteAffiliateBps,
-  (calculatedFees, affiliateBps) => {
-    if (!affiliateBps) return
-    if (affiliateBps === '0') return bn(0)
-
-    return calculatedFees.feeUsd
   },
 )
 
-export const selectTradeQuoteAffiliateFeeDiscountUsd = createSelector(
-  (state: ReduxState) =>
-    selectCalculatedFees(state, {
-      feeModel: 'SWAPPER',
-      inputAmountUsd: selectQuoteSellAmountUsd(state),
-    }),
+export const selectTradeAffiliateFeeUsd = createSelector(
+  selectQuoteSellAmountUsd,
   selectActiveQuoteAffiliateBps,
-  (calculatedFees, affiliateBps) => {
+  (sellAmountUsd, affiliateBps) => {
+    const { feeUsd } = calculateFeeUsd({
+      inputAmountUsd: bnOrZero(sellAmountUsd),
+    })
+
     if (!affiliateBps) return
     if (affiliateBps === '0') return bn(0)
 
-    return calculatedFees.foxDiscountUsd
+    return feeUsd
   },
 )
 
 export const selectTradeQuoteAffiliateFeeAfterDiscountUserCurrency = createSelector(
-  selectTradeQuoteAffiliateFeeAfterDiscountUsd,
-  selectUserCurrencyToUsdRate,
-  (tradeAffiliateFeeAfterDiscountUsd, sellUserCurrencyRate) => {
-    if (!tradeAffiliateFeeAfterDiscountUsd || !sellUserCurrencyRate) return
-    return bn(tradeAffiliateFeeAfterDiscountUsd).times(sellUserCurrencyRate).toFixed()
-  },
-)
-
-export const selectTradeQuoteAffiliateFeeDiscountUserCurrency = createSelector(
-  selectTradeQuoteAffiliateFeeDiscountUsd,
+  selectTradeAffiliateFeeUsd,
   selectUserCurrencyToUsdRate,
-  (tradeAffiliateFeeDiscountUsd, sellUserCurrencyRate) => {
```

Comment:
Naming was wrong and confusing, this really meant `tradeAffiliateFeeAfterDiscountUsd ` which now can be simplified as `tradeAffiliateFeeUsd`